design-documents/graph-ql/coverage/company.md

```diff
@@ -0,0 +1,527 @@
+    items: [CompanyRole] @doc(description: "An array of 'CompanyUser' objects that match the specified search criteria.")
+    total_count: Int @doc(description: "The number of objects returned.")
+    page_info: SearchResultPageInfo @doc(description: "An object that includes the pagination meta data of the search query.")
+}
+
+type CompanyRole @doc(description: "Company role output data schema returned in response to a query by Role ID.") {
+    id: Int @doc(description: "Role ID.")
+    name: String @doc(description: "Role name.")
+    users_count: Int @doc(description: "Total number of Users with such Role within Company Hierarchy.")
+    permissions: [String] @doc(description: "A list of permission resources defined for a Role.")
+}
+
+type CompanyAclResource @doc(description: "Output data schema for an object with Role permission resource information.") {
+    id: String @doc(description: "ACL resource ID.")
+    text: String @doc(description: "ACL resource label.")
+    sortOrder: Int @doc(description: "ACL resource sort order.")
+    children: [CompanyAclResource!] @doc(description: "An array of sub-resources.")
+}
+
+type CompanyUserEmailCheckResponse @doc(description: "Response object schema for a Company User email validation query.") {
+    status: Int @doc(description: "1 - 'OK', 0 - 'Error'")
+    message: String @doc(description: "Error or notice message for a User.")
+}
+
+type CompanyHierarchyOutput @doc(description: "Response object schema for a Company Hierarchy query.") {
+    structure: CompanyHierarchyElement @doc(description: "An array of Company structure elements.")
+    draggable: Boolean @doc(description: "Flag that defines whether Company Hierarchy can be changed by current User or not.")
+    max_nesting: Int @doc(description: "Indicator of maximun nesting of elements within a whole Company Hierarchy.")
+}
+
+type CompanyHierarchyElement @doc(description: "Company Hierarchy element output data schema.") {
+    id: Int @doc(description: "Hierarchy element ID.")
+    tree_id: Int @doc(description: "The hierarchical ID of the element within a structure. Used for changing element's position in hierarchy.")
+    type: String @doc(description: "Hierarchy element type: 'customer' or a 'team'.")
+    text: String @doc(description: "Hierarchy element name.")
+    description: String @doc(description: "Hierarchy element description.")
+    opened: Boolean @doc(description: "Flag that defines whether Hierarchy element should be expanded or not.")
+    children: [CompanyHierarchyElement!] @doc(description: "An array of child elements.")
+}
+
+type CompanyTeam @doc(description: "Company Team entity output data schema.") {
+    id: Int @doc(description: "Team ID.")
+    name: String @doc(description: "Team name.")
+    description: String @doc(description: "Team description.")
+}
+
+type SearchResultPageInfo @doc(description: "Output data schema for a meta data object of the search query response.") {
```

Comment:
Let's not include existing types to the schema, we can mention that these are existing types.

Reply:
fixed